benchmarks/compile_benchmark.sh

```diff
@@ -0,0 +1 @@
+clang++ -o benchmark_nanorange -Wall -Werror --std=c++14 -O3 -I ../include -lbenchmark benchmark_main.cpp $1
```

Comment:
I'd rather use CMake to (optionally) build this, rather than using a shell script. If you don't fancy writing the necessary CMakeLists stuff then just remove this file and I'll add it after merging the PR.

Reply:
Thanks, I experimented a bit with CMake - but I don't really know it, so I left it as is for me.

What did you think about my point of 'building benchmarks independently'?
Alternative, by the way, could be to add "-mllvm -align-all-blocks=5" - well at least that was the suggestion of google benchmark people. This will force icache alignment.

Reply:
My preference would be for one executable per algorithm, so we could have `benchmark_rotate.exe`, `benchmark_sort.exe`, `benchmark_nth_element.exe` etc. Would that avoid the problems you mentioned?

Reply:
No, not really. But it's a reasonable thing to do - libc++ does that for example.

You'll see it when you hit it - the difference would be dramatic and a little bit of tweaking will make it go away. Keeping the benchmarks in individual executables will help with working on it - at least tweaking one won't affect the other.